ducktape/tests/result.py

```diff
@@ -34,41 +33,40 @@ def __str__(self):
 class TestResult(object):
     """Wrapper class for a single result returned by a single test."""
 
-    def __init__(self, test_context, test_status=PASS, summary="", data=None, start_time=-1, stop_time=-1):
+    def __init__(self,
+                 test_context,
+                 session_context,
+                 test_status=PASS,
+                 summary="",
+                 data=None,
+                 start_time=-1,
+                 stop_time=-1):
         """
         @param test_context  standard test context object
         @param test_status   did the test pass or fail, etc?
         @param summary       summary information
         @param data          data returned by the test, e.g. throughput
         """
 
-        self.test_context = test_context
-        self.session_context = self.test_context.session_context
+        self.test_id = test_context.test_id
```

Comment:
yeah - Test context isn't currently serializable, so we can't have a reference to a test_context object if we want to serialize TestResult